packages/venia-ui/lib/components/MyAccount/myAccount.js

```diff
@@ -1,29 +1,21 @@
-import React, { useCallback } from 'react';
+import React from 'react';
 import { Archive as HistoryIcon, LogOut as SignOutIcon } from 'react-feather';
 import { func, shape, string } from 'prop-types';
 
 import { mergeClasses } from '../../classify';
 import AccountLink from './accountLink';
 import defaultClasses from './myAccount.css';
-
-const DEFAULT_TITLE = 'My Account';
-const UNAUTHED_TITLE = 'Signing Out';
-const UNAUTHED_SUBTITLE = 'Please wait...';
+import useMyAccount from '@magento/peregrine/lib/mixins/MyAccount/useMyAccount';
 
 const PURCHASE_HISTORY = 'Purchase History';
 const SIGN_OUT = 'Sign Out';
 
 const MyAccount = props => {
-    const { signOut, user } = props;
-    const { email, firstname, lastname } = user;
-    const name = `${firstname} ${lastname}`.trim() || DEFAULT_TITLE;
-    const title = email ? name : UNAUTHED_TITLE;
-    const subtitle = email ? email : UNAUTHED_SUBTITLE;
     const classes = mergeClasses(defaultClasses, props.classes);
 
-    const handleSignOut = useCallback(() => {
-        signOut({ history: window.history });
-    }, [signOut]);
+    const { handleSignOut, subtitle, title } = useMyAccount({
```

Comment:
Some mixins are going to return a lot of props, and like this one, they may take a lot of arguments as well. Destructuring is kind of tough to read in those scenarios, so I recommend splitting it into two statements.

```es6
const mixinProps = useMyAccount({
  multiline: "really",
  long: "arguments",
  make: "it hard to read"
})
const {
  handleSignOut,
  subtitle,
  title: renamedTitle // avoid clash with variable in closure
} = mixinProps
```

Reply:
Sure. I like this idea.